clickhouse/columns/array.cpp

```diff
@@ -5,43 +5,58 @@
 namespace clickhouse {
 
 ColumnArray::ColumnArray(ColumnRef data)
+    : Column(Type::CreateArray(data->Type()))
+    , data_(data->CloneEmpty())
+    , offsets_(std::make_shared<ColumnUInt64>())
+{
+}
+
+ColumnArray::ColumnArray(ColumnRef data, ShareOwnershipTag)
     : Column(Type::CreateArray(data->Type()))
     , data_(data)
     , offsets_(std::make_shared<ColumnUInt64>())
 {
 }
 
+// Explicitly sharing `data_` column is required by ColumnArrayT
+ColumnArray::ColumnArray(ColumnArray&& other)
+    : Column(other.Type())
+    , data_(std::move(other.data_))
+    , offsets_(std::move(other.offsets_))
+{
+}
+
```

Comment:
What do you think about leaving a comment mentioning that these exist because of the ColumnArrayT?